src/gui/owncloudgui.cpp

```diff
@@ -347,6 +348,15 @@ void ownCloudGui::addAccountContextMenu(AccountStatePtr accountState, QMenu *men
     actionOpenoC->setProperty(propertyAccountC, QVariant::fromValue(accountState->account()));
     QObject::connect(actionOpenoC, SIGNAL(triggered(bool)), SLOT(slotOpenOwnCloud()));
 
+    if (accountState->state() == AccountState::Connected) {
+        menu->addSeparator();
+
+        QString quotaInfo = tr("Fetching Quota");
+        auto actionQuotaInfo = menu->addAction(quotaInfo);
+        actionQuotaInfo->setEnabled(false);
+        new QuotaAction(actionQuotaInfo, accountState);
```

Comment:
@rullzer You make a new QuotaAction and thereby QuotaInfo every time the menu is updated, resetting QuotaInfo's 30s timer. So there will be a request for each account every time the menu is updated, and every 30s after that. This is in addition to the quota checks that happen when the settings are open.

I think tray menu regenerations are currently on-demand or max-every-30s. That's probably why it looks similar.

I think there should be one thing (probably QuotaInfo) gathering quota information regularly, and other components should make use of that information. Instantiating another quota-retriever to get that data into the tray sets a wrong precedent even if it's probably few enough requests to not matter much.